From vitb@kernel.crashing.org Wed Sep 26 13:36:38 2007
From: Vitaly Bordug <vitb@kernel.crashing.org>
Date: Thu, 27 Sep 2007 00:36:22 +0400
Subject: USB: serial gadget: Disable endpoints on unload
To: Greg KH <greg@kroah.com>
Cc: linuxppc-dev@ozlabs.org
Cc: linuxppc-dev@ozlabs.org
Message-ID: <20070926203622.24118.36016.stgit@localhost.localdomain>



After Serial gadget is being unloaded, neither serial itself, nor other
gadget stuff can be loaded subsequently.

Signed-off-by: Vitaly Bordug <vitb@kernel.crashing.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/serial.c |    6 ++++++
 1 file changed, 6 insertions(+)

--- a/drivers/usb/gadget/serial.c
+++ b/drivers/usb/gadget/serial.c
@@ -1470,6 +1470,12 @@ static void /* __init_or_exit */ gs_unbi
 			dev->dev_ctrl_req = NULL;
 		}
 		gs_free_ports(dev);
+		if (dev->dev_notify_ep)
+			usb_ep_disable(dev->dev_notify_ep);
+		if (dev->dev_in_ep)
+			usb_ep_disable(dev->dev_in_ep);
+		if (dev->dev_out_ep)
+			usb_ep_disable(dev->dev_out_ep);
 		kfree(dev);
 		set_gadget_data(gadget, NULL);
 	}
